feature_tests/src/structs.rs

```diff
@@ -62,10 +71,78 @@ pub mod ffi {
         }
     }
 
+    impl OtherOpaque {
+        pub fn from_usize(number: usize) -> Box<OtherOpaque> {
+            Box::new(OtherOpaque(Mutex::new(format!("{number}"))))
+        }
+
+        pub fn change(&self, number: usize) {
+            let mut guard = self.0.lock().expect("Failed to lock mutex");
+            *guard = format!("{number}");
+        }
+
+        #[allow(clippy::needless_lifetimes)]
+        pub fn borrow<'a>(&'a self) -> &'a OtherOpaque {
```

Comment:
nit: everything returning opaque references will need  `#[diplomat::skip_if_ast]` for CI to pass.